### PR TITLE
[TR_238] Map View

### PR DESCRIPTION
**Pull request by duskcloudxu**
---

#### Please check if the PR fulfills these requirements

- [ ] The changes don't come from your master branch. ([Source](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)) 

---

#### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Map Screen, and related components

#### What is the new behavior? (if this is a feature change)

Kapture 2020-07-26 at 13 58 22

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
No

#### Images (before/ after screenshots, interaction GIFs, ...)

---

#### TODO

- Not sure about the reason, but the map has low performance once the donationMarkers are loaded, would try to find reason later

- Add navigation function to the donationMarkers once the claimDetailScreen is done.

- The current UI is not as good as design due to distribution of donations are too close. Maybe we should update the design or change UI of donationMarkers when map scale is too large( like change into a dot or something)

### CODE REVIEW

**Comment by sagehen03** [comment]
Hey Tao,
It looks like these changes make it so that we can no longer test via a browser.  I understand that the map is not supported for web, but it would be nice if everything except for the map continued to work for web since that's the fastest way to test I think.
image

File: src/screens/MapScreen/MapScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			>
+				{(donations as any).map(item => {
+					const { id, food_name } = item;
+					const distance = 42;
+					return (
+						<DonationMarker
+							coordinate={{
+								latitude: item.donor.latitude,
+								longitude: item.donor.longitude,
+							}}
```
Comment (sagehen03) [inline src/screens/MapScreen/MapScreen.tsx]: When I tested on my phone with the expo app, I got some errors in my console and the app crashed when I tried to toggle from map to list view.  The suggestion below seems to help in my case.
```suggestion
							key={id}
							coordinate={{
								latitude: parseFloat(item.donor.latitude),
								longitude: parseFloat(item.donor.longitude),
							}}
```

**Comment by sagehen03** [comment]
It seems like switching between map and list view is very slow.  Sometimes when I click on the list button, it never changes from the map view. Other times it changes after more than 30 seconds.  I'm going to look into this a bit more.

**Comment by duskcloudxu** [comment]
Hi Drew, I optimized the code so it could work without performance problems now...also I made the markers change the size according to the map scale, it might look messy now since many donors using the same geodata and we could optimize it later by changing the position of donors in the database.

For the problem that the introduction of mapview causes web browser version to crush, I made some research and seems it's a bug that has not been solved yet(https://github.com/expo/expo/issues/8369). I would say that the most promising way is to add a dummy map component on the web browser, how do you think about it?

**Comment by duskcloudxu** [comment]
> Hi Drew, I optimized the code so it could work without performance problems now...also I made the markers change the size according to the map scale, it might look messy now since many donors using the same geodata and we could optimize it later by changing the position of donors in the database.
> 
> For the problem that the introduction of mapview causes web browser version to crush, I made some research and seems it's a bug that has not been solved yet([expo/expo#8369](https://github.com/expo/expo/issues/8369)). I would say that the most promising way is to add a dummy map component on the web browser, how do you think about it?

Oh and I just found mentioning it would add our thread into that issue lol, would that be a problem?

**Review by sagehen03 — APPROVED** [review]
seems to be working now.  I'm going to merge.

**Pull request merged**